src/library/rekordbox/rekordboxfeature.cpp

```diff
@@ -163,7 +163,7 @@ bool dropTable(QSqlDatabase& database, const QString& tableName) {
 
 // This function is executed in a separate thread other than the main thread
 // The returned list owns the pointers, but we can't use a unique_ptr because
-// the result is passed by a const reference
+// the result is passed by a const reference inside QFuture.
```

Comment:
thank you for updating the comment. It still didn't make sense for me until I dug a little deeper. As far as I can tell the actual problem is that QFuture needs to be parameterized on a type that has copy constructors? If so, that also highlights a potential safety issue. If the QList is copyable and it just copies the contained pointers, that would lead to dangling pointers with double free's etc. Right?

Reply:
>  QFuture needs to be parameterized on a type that has copy constructors

Yes. 

> If so, that also highlights a potential safety issue. If the QList is copyable and it just copies the contained pointers, that would lead to dangling pointers with double free's etc. Right?

Dangling pointers are freed pointers, that are not null. This does not happens in this case, because the QList only owns the pointers not the object. When the copy in the worker thread is deleted only the memory location of the pointers is deleted, not the objects the have been pointing to. They are accessed in the main thread and made owning and managed by the `std::vector<std::unique_ptr>>` before second copy of the original QList is deleted.

Reply:
Comment updated.

Reply:
> When the copy in the worker thread is deleted only the memory location of the pointers is deleted, not the objects the have been pointing to.

Ah, I assumed the QList was deleting the objects pointed to the pointers, that makes sense.